src/table/table.component.ts

```diff
@@ -248,30 +248,79 @@ export interface TableTranslations {
+	 * Provide the initial model if you have data that is not completely loaded and set the table's model to the generated skeleton model.
+	 * When the data is fully loaded switch back to the original model which now contains the fully loaded data.
+	 *
+	 *	Example:
+	 *
+```typescript
+model = new TableModel();
+this.model.header = [
+	new TableHeaderItem({
+		data: "Name"
+	}),
+	new TableHeaderItem({
+		data: "hwer",
+		className: "my-class"
+	})
+];
+
+this.model.rowsSelectedChange.subscribe(event => console.log(event));
+
+this.model.data = [
+	[new TableItem({ data: "Name 1" }), new TableItem({ data: "qwer" })],
+	[new TableItem({ data: "Name 3" }), new TableItem({ data: "zwer" })],
+	[new TableItem({ data: "Name 2" }), new TableItem({ data: "swer" })],
+	[new TableItem({ data: "Name 4" }), new TableItem({data: "twer"})],
+	[new TableItem({ data: "Name 5" }), new TableItem({data: "twer"})],
+	[new TableItem({ data: "Name 6" }), new TableItem({data: "twer"})],
+	[new TableItem({ data: "Name 7" }), new TableItem({data: "twer"})]
+];
+// Note that if you provide the skeletonModel with an initial tableModel it makes the row and column count parameter not do anything.
+const skeletonModel = Table.skeletonModel(0, 0, this.model);
+// Set the table model to `skeletonModel` and switch back to `this.model` when data is finished loading.
+```
+	 *
 	 */
-	static skeletonModel(rowCount: number, columnCount: number) {
-		const model = new TableModel();
-		let header = new Array<TableHeaderItem>();
-		let data = new Array<Array<TableItem>>();
-		let row = new Array<TableItem>();
-
-		for (let i = 0; i < columnCount; i++) {
-			header.push(new TableHeaderItem());
-			row.push(new TableItem());
+	static skeletonModel(rowCount: number, columnCount: number, initialModel?: TableModel) {
+		// If no initial model is supplied create a model with empty data.
+		if (!initialModel) {
+			const model = new TableModel();
+			let header = new Array<TableHeaderItem>();
+			let data = new Array<Array<TableItem>>();
+			let row = new Array<TableItem>();
+
+			for (let i = 0; i < columnCount; i++) {
+				header.push(new TableHeaderItem());
+				row.push(new TableItem({ data: " " }));
+			}
+
+			for (let i = 0; i < rowCount - 1; i++) {
+				data.push(row);
+			}
+
+			model.header = header;
+			model.data = data;
+			return model;
 		}
-		for (let i = 0; i < rowCount - 1; i++) {
-			data.push(row);
+		// If an initial model is provided, to create a skeleton state from the given model,
+		// create only first row with empty data to show loading animation only on the first row.
+		const model = Object.assign( Object.create( Object.getPrototypeOf(initialModel)), initialModel);
+		model.data[0] = model.data[0].map(col => new TableItem());
+
+		for (let i = 1; i < model.data.length; i++) {
+			model.data[i] = model.data[i].map(col => new TableItem({ data: " " }));
```

Comment:
We should reuse the templates from the provided model so they can just add the data to the new model.
Doesn't look like it costs us anything to allow that.